opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -1442,4 +1442,35 @@ namespace Opm
     }
 
 
+
+
+
+
+    template<typename TypeTag>
+    WellControls*
+    WellInterface<TypeTag>::
+    createWellControlsWithBHPAndTHP(DeferredLogger& deferred_logger) const
+    {
```

Comment:
This function is in this way to make sure this PR is only a refactoring PR to limit the code that actually changes behavoir.  This function will be changed in the following PRs depending how we will use it.